From da5c4d13f1acaba98ac601e973d621c72eb5e9c6 Mon Sep 17 00:00:00 2001
From: Aloka Dixit <quic_alokad@quicinc.com>
Date: Thu, 3 Mar 2022 15:22:46 -0800
Subject: [PATCH 4/7] mac80211: reset puncturing bitmap during width downgrade

Set the RU puncturing bitmap to 0 if for any reason bandwidth
is downgraded to less than 80 Mhz.

Signed-off-by: Aloka Dixit <quic_alokad@quicinc.com>
---
 net/mac80211/util.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/net/mac80211/util.c b/net/mac80211/util.c
index 3aade17ea6c4..4b0219925f33 100644
--- a/net/mac80211/util.c
+++ b/net/mac80211/util.c
@@ -4090,6 +4090,13 @@ u32 ieee80211_chandef_downgrade(struct cfg80211_chan_def *c)
 		break;
 	}
 
+	if ((c->width != NL80211_CHAN_WIDTH_320) &&
+	    (c->width != NL80211_CHAN_WIDTH_160) &&
+	    (c->width != NL80211_CHAN_WIDTH_80)) {
+		c->ru_punct_bitmap = 0;
+		c->ru_punct_bitmap_supp_he = 0;
+	}
+
 	WARN_ON_ONCE(!cfg80211_chandef_valid(c));
 
 	return ret;
-- 
2.31.1

